### PR TITLE
token gating, channel info design

### PR DESCRIPTION
**Pull request by bokuhe**
feat
- check user balance when setting token gating
- added a button to clear when explore search is entered

design
- channel info

### CODE REVIEW

File: src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx [hunk 1]
```diff
@@ -56,152 +60,204 @@ const TokenGating = ({
...
+
+      setEnableDoneButton(input > 0 && count >= input)
+    } else {
+      // Native
+      const balance = UTIL.toBn(userBalances.balance)
+      const input = UTIL.toBn(UTIL.microfyP(gatingTokenAmount as Token))
+
+      setEnableDoneButton(!input.isZero() && balance.gte(input))
+    }
+  }, [gatingTokenAmount, userBalances, userNfts])
```
Comment (denniswon) [inline src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx:99]: why is this needed? Also, for performance, can we make this to query userBalances or userNfts depending on the token gating type?

**Review by denniswon — CHANGES_REQUESTED** [review]

File: src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx (hunk 1 above)
Comment (bokuhe) [reply src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx:99]: https://klaytn-foundation.atlassian.net/browse/WEB3-198

The view is not separated, and token selection and value input are being done at the same view. 
I will refactor this.

File: src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx [hunk 2]
```diff
@@ -56,152 +60,204 @@ const TokenGating = ({
     limit: 30,
   })
 
+  const userBalances = useUserBalance({
```
[the remark this answers is not in the document]
Comment (bokuhe) [reply src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx:63]: renamed

File: src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx [hunk 3]
```diff
@@ -56,152 +60,204 @@ const TokenGating = ({
...
-        }
-        onPressLeft={(): void => {
-          if (step === 2) {
-            setStep(1)
+    <Container style={styles.container}>
+      <KeyboardAvoidingView
+        behavior={Platform.select({ ios: 'padding' })}
+        style={{ flex: 1 }}
+        keyboardVerticalOffset={bottom + 12}
+      >
```
[the remark this answers is not in the document]
Comment (bokuhe) [reply src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx:107]: Container view's keyboardAvoiding doesn't work.

**Review by denniswon — APPROVED** [review]

**Comment by denniswon** [comment]
much better. thanks!
